src/actions/apiCalls.ts

```diff
@@ -0,0 +1,79 @@
+import { ApiEndpoint } from "../api";
+import ApiCallError from "../errors/ApiCallError";
+import { AsyncAction } from ".";
+
+export const enum ApiCallsActionType {
+  STARTED = "STARTED",
+  SUCCESS = "SUCCESS",
+  FAILURE = "FAILURE"
+}
+
+export interface ApiCallsActionBase {
+  endpoint: ApiEndpoint;
+  identifier: string;
+}
+
+interface ApiCallStartedAction extends ApiCallsActionBase {
+  type: ApiCallsActionType.STARTED;
+}
+interface ApiCallSuccessAction extends ApiCallsActionBase {
+  type: ApiCallsActionType.SUCCESS;
+}
+interface ApiCallFailedAction extends ApiCallsActionBase {
+  type: ApiCallsActionType.FAILURE;
+  error: ApiCallError;
+}
+export type ApiCallsAction =
+  | ApiCallStartedAction
+  | ApiCallSuccessAction
+  | ApiCallFailedAction;
+
+/**
+ * Wrap an action that runs an API call, so that:
+ * a) Errors from failed API calls are caught, and
+ * b) The state of the API request is logged.
+ *
+ * @param asyncAction API call redux action to be wrapped
+ * @param endpoint API endpoint being hit
+ * @param identifier A way of identifying the outgoing API call. For example, if
+ * the logged in user calls the API to trust another user, that other user's UID
```

Comment:
I'd also note that `identifier` should be unique if the api call is not idempotent. It's not an issue for `trust` and `requestInvite`, but using `UID` for a `pay` call would probably be a bad idea.